lib/node_modules/@stdlib/random/base/mt19937/lib/factory.js

```diff
@@ -88,21 +101,53 @@ function factory( seed ) {
 		mt[ mti ] >>>= 0;
 	}
 
+	if ( seedIsArray ) {
+		for (; k > 0; k--) {
+			s = mt[ i - 1 ] ^ ( mt[ i - 1 ] >>> 30 );
+
+			mt[i] = (mt[i] ^ (((((s & 0xffff0000) >>> 16) * 1664525) << 16) +
+				( (s & 0x0000ffff) * 1664525 ) ) ) + seed[ j ] + j;
+			mt[i] >>>= 0;
+			i += 1;
+			j += 1;
+			if (i >= N) {
+				mt[0] = mt[N - 1];
+				i = 1;
+			}
+			if (j >= seed.length) {
+				j = 0;
+			}
+		}
+
+		for (k = N - 1; k; k--) {
+			s = mt[i - 1] ^ (mt[i - 1] >>> 30);
+			mt[i] = (mt[i] ^ (((((s & 0xffff0000) >>> 16) * 1566083941) << 16) +
+				( (s & 0x0000ffff ) * 1566083941 ) ) ) - i;
+			mt[i] >>>= 0;
+			i += 1;
+			if (i >= N) {
+				mt[0] = mt[N - 1];
+				i = 1;
+			}
+		}
+		mt[0] = 0x80000000;
+	}
+
 	setReadOnly( mt19937, 'NAME', 'mt19937' );
 	setReadOnly( mt19937, 'SEED', state );
 	setReadOnly( mt19937, 'MIN', 1 );
-	setReadOnly( mt19937, 'MAX', UINT32_MAX - 1 );
+	setReadOnly( mt19937, 'MAX', UINT32_MAX );
 	setReadOnly( mt19937, 'normalized', normalized );
 
 	setReadOnly( normalized, 'NAME', mt19937.NAME );
 	setReadOnly( normalized, 'SEED', mt19937.SEED );
-	setReadOnly( normalized, 'MIN', ( mt19937.MIN - 1.0 ) / NORMALIZATION_CONSTANT );
-	setReadOnly( normalized, 'MAX', ( 4294967294.0 ) * ( 1.0 / 4294967295.0 ) );
+	setReadOnly( normalized, 'MIN', ((( mt19937.MIN - 1.0 )) >>> 0) / ( NORMALIZATION_CONSTANT >>> 0 ) );
```

Comment:
What is the rationale for ensuring `uint32` operands here?

Reply:
Wait...when you do

```javascript
( mt19937.MAX - 1.0 ) / NORMALIZATION_CONSTANT
```
you get back an integer value? Something seems off. The above, if `MAX` is `4294967294.0`, should return a value between `0` and `1`. If it returns an integer, something seems amiss. Can you provide the exact code which generates `-(UINT32_MAX-1)`?

Reply:
~~~~
setReadOnly( normalized, 'MAX', ((( mt19937.MAX - 1.0 )) ) / ( NORMALIZATION_CONSTANT  ) );
~~~~

This gives the error shown earlier.